moduleframework/helpers/openshift_helper.py

```diff
@@ -0,0 +1,300 @@
+        :return:
+        """
+        if env:
+            oc_ip = common.get_openshift_ip()
+            oc_user = common.get_openshift_user()
+            oc_passwd = common.get_openshift_passwd()
+        oc_output = self.runHost("oc login %s:8443 --username=%s --password=%s" % (oc_ip,
+                                                                                   oc_user,
+                                                                                   oc_passwd),
+                                 verbose=common.is_not_silent())
+        return oc_output.exit_status
+
+    def tearDown(self):
+        """
+        Cleanup environment and call also cleanup from config
+
+        :return: None
+        """
+        super(OpenShiftHelper, self).tearDown()
+        try:
+            self._app_remove()
+        except Exception as e:
+            common.print_info(e, "OpenShift application already removed")
+            pass
+
+    def _get_ip_instance(self):
+        """
+        It gets and IP address of an application from of OpenShift POD.
+        :return: True: getting IP address was successful
+                 False: getting IP address was not successful
+        """
+        oc_get_service = self.runHost("oc get service -o json")
+        service = self._convert_string_to_json(oc_get_service.stdout)
+        try:
+            for svc in service:
+                if svc.get('metadata').get('labels').get('app') == self.app_name:
+                    self.ipaddr = svc.get('spec').get("clusterIP")
+                    common.trans_dict['GUESTIPADDR'] = self.ipaddr
+            return True
+        except KeyError as e:
+            common.print_info(e.message)
+            return False
+        except IndexError as e:
+            common.print_info(e.message)
+            return False
+
+    def getIPaddr(self):
+        """
+        Return protocol (IP or IPv6) address on a POD OpenShift instance.
+
+        It returns IP address of POD instance
+
+        :return: str
+        """
+        return self.ipaddr
+
+    def start(self):
+        """
+        starts the OpenShift application
+
+        :param args: Do not use it directly (It is defined in config.yaml)
+        :param command: Do not use it directly (It is defined in config.yaml)
+        :return: None
+        """
+        if not self._app_exists():
+            self._create_app()
+            # Verify application is really deploy and prepared for testing.
+            self._verify_pod()
+        self._get_ip_instance()
+
+    def stop(self):
+        """
+        Stops the OpenShift
+
+        :return: None
+        """
+        if self.status():
+            try:
+                self._app_remove()
+            except Exception as e:
+                common.print_info(e, "OpenShift application already removed")
+                pass
+
+    def status(self):
+        """
+        get status of an application in OpenShift environment
```

Comment:
I still think that this docstring doesn't correspond with the code: what does the "status" mean exactly? For me, status is usually enum, something like "running", "failed", "finished" and this method returns bool. It makes me confused.

Reply:
This should be solved in whole MTF. `status` function always return `bool` value like e.g. here: https://github.com/fedora-modularity/meta-test-family/blob/devel/moduleframework/common.py#L552
https://github.com/fedora-modularity/meta-test-family/blob/devel/moduleframework/helpers/container_helper.py#L171

I have filed an issue for it here: https://github.com/fedora-modularity/meta-test-family/issues/168

I will keep it now as it is.